### PR TITLE
Popover: rewrite Storybook examples using controls

### PR DESCRIPTION
**Pull request by ciampo**
## What?

In the context of #42770, the Storybook examples are being of little help when working on the `Popover` component in isolation.

This PR refactors the Storybook examples to use controls (instead of the deprecated knobs add-on), and also rewrites some of the stories in a way that hopefully showcases the usages of the component in a better way compared to how it was before.

## Why?

Better Storybook examples = better dev experience (both as a component author, and as a component consumer).

## How?

1. Converted from knobs to controls, listing all known `Popover` props
2. The `default` example has been changed to mirror the example snippet in the component's README (see notes below). I feel like the previous default example was not really useful anyway.
3. The `positions` example has been replaced with a new story that showcases all possible values for `placement` (ie. the new prop that is supposed to replace `position`)
4. The dragging example has been removed, as I feel it didn't give much value after adding the new "default" and "all placements" story (and its code was very complex)
5. The "Dynamic height" story has been kept mostly intact
6. Added a new story to test the behavior of the popover when its anchor is inside an iframe, but the popover itself is rendered outside of that iframe

## Testing Instructions

1. Make sure that Storybook builds correctly and that all stories are accessible
2. Make sure that 

### Known issues

Working on this refactor highlighted a couple of issues:

####  The `Popover` is not anchoring itself correctly to its parent (which is the default behaviour when an explicit anchor it not passed):

https://user-images.githubusercontent.com/1083581/182440756-1d02165f-b628-45e8-9b42-7370a7e4f050.mp4

This behavior can be also spotted on `trunk` by visiting [the positioning storybook example](https://wordpress.github.io/gutenberg/?path=/story/components-popover--positioning), and noting that the `Popover` is not positioning itself correctly with respect to its parent (the pink bordered box). This is likely another regression introduced by #40740, and if folks confirm it, I will create a new issue and add it to #42770

#### The positioning of the `Popover` is a bit off in the `iframe` example

https://user-images.githubusercontent.com/1083581/182441479-25761b05-6b4d-40d6-a3c1-4b4352f74471.mp4

I haven't had time to debug this example in depth yet, but it looks like the offset applied to the `Popover` to compensate the `iFrame` positioning is not correct. It's interesting (and worrying) that the `Popover` doesn't work as expected with a simple `iframe` example. We should understand if something is wrong in the way I set up the Storybook example, or if this is another bug that we need to add to #42770 and fix.

### CODE REVIEW

File: packages/components/src/popover/index.js [hunk 1]
```diff
@@ -162,25 +162,26 @@ const Popover = (
 		: placement;
 
 	const ownerDocument = useMemo( () => {
+		let documentToReturn;
```
Comment (ciampo) [inline packages/components/src/popover/index.js]: These changes are temporarily applied to this PR, while waiting for #42886 to get merged

File: packages/components/src/popover/stories/index.js [hunk 2]
```diff
@@ -1,123 +1,182 @@
...
 /**
  * Internal dependencies
  */
-import { DraggableWrapper } from './_utils';
-import Popover from '../';
 import Button from '../../button';
+import { Provider as SlotFillProvider } from '../../slot-fill';
+import Popover from '../';
+
+// Format: "[yAxis] [xAxis]"
```
Comment (ciampo) [inline packages/components/src/popover/stories/index.js:14]: It actually looks like the `position` prop accepted also a third value, called `corner` — although I couldn't find any reference in the old popover documentation  and I didn't have enough time to investigate deeper

File: packages/components/src/popover/style.scss [hunk 3]
```diff
@@ -73,9 +73,9 @@ $arrow-triangle-base-size: 14px;
 		content: "";
 		position: absolute;
 		top: -1px;
-		left: 0;
+		left: 1px;
 		height: 2px;
-		width: 100%;
+		right: 1px;
```
Comment (ciampo) [inline packages/components/src/popover/style.scss:78]: Happy to move this adjustment to a separate PR if necessary

File: packages/components/src/popover/stories/index.js [hunk 4]
```diff
@@ -1,123 +1,182 @@
...
+			options: AVAILABLE_POSITIONS,
+		},
+		__unstableSlotName: { control: { type: null } },
+		__unstableObserveElement: { control: { type: null } },
+		__unstableForcePosition: { control: { type: 'boolean' } },
+		__unstableShift: { control: { type: 'boolean' } },
+	},
 	parameters: {
-		knobs: { disable: false },
+		docs: { source: { state: 'open' } },
```
Comment (mirka) [inline packages/components/src/popover/stories/index.js]: I think we can keep the code snippets collapsed for these stories, since they're not really useful in the current state. (Maybe with the exception of the iframe one?)

**Review by mirka — APPROVED** [review]
A much needed improvement for the intense Popover work going on. Thanks!

File: packages/components/src/popover/stories/index.js (hunk 4 above)
Comment (ciampo) [reply packages/components/src/popover/stories/index.js]: I removed the whole `parameters` key for now in https://github.com/WordPress/gutenberg/pull/42903/commits/0381cd8e30d45392811a80190835cd1805ca4047. I'm sure we'll iterate over this file soon enough, especially when we'll convert this component to TypeScript

File: packages/components/src/popover/index.js (hunk 1 above)
Comment (ciampo) [reply packages/components/src/popover/index.js]: https://github.com/WordPress/gutenberg/pull/42886 got merged, and I removed those changes
